capstone/src/main/webapp/youtubeTypescript.ts

```diff
@@ -246,21 +251,18 @@ async function getTrending(categoryId: number) {
   });
 }
 
-/** Enables user from entering text into the text field */
-function enableTextInput(button) {
+/** Enables and disables manual input into the text field */
+function textInputToggle (button, toEnable: boolean) {
   if (button.checked) { 
-    (<HTMLInputElement>document.getElementById('channelIdForAnalysis')).value = button.id;
-    (<HTMLInputElement>document.getElementById('channelIdForAnalysis')).disabled = true;
-    (<HTMLInputElement>document.getElementById("keywordSearch")).disabled = true;
-  }
-}
-
-/** Disables user from entering text into the text field */
-function disableTextInput(button) {
-  if (button.checked) { 
-    (<HTMLInputElement> document.getElementById('channelIdForAnalysis')).value = "";
-    (<HTMLInputElement>document.getElementById('channelIdForAnalysis')).disabled = false;
-    (<HTMLInputElement>document.getElementById("keywordSearch")).disabled = false;   
+    if (toEnable) {
+      (<HTMLInputElement> document.getElementById('channelIdForAnalysis')).value = button.id;
```

Comment:
Just a suggestion, but possibly you can create a function like `getInputElement(id: string)` which returns a HTMLInputElement?  Then you might be able to write more concise code that doesn't have all the repetative `<HTMLInputElement>` casts?  Maybe something like:
```
let channelIdEl = getInputElement('channelIdForAnalysis');
channelIdEl.value = button.id;
...
```

Reply:
addressed in "add getInputElement function'